cdap-proto/src/main/java/io/cdap/cdap/proto/credential/CredentialProfile.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.credential;
+
+import java.util.Map;
+
+/**
+ * Defines a profile for credential provisioning.
+ */
+public class CredentialProfile {
+
+  private final String provisionerType;
```

Comment:
what would you think about changing the naming from credential provisioner to credential provider? It would match the current OAuth handler terminology more closely and also be more clearly different than cluster provisioners

Reply:
Sounds reasonable, changed